arch/xtensa/core/thread.c

```diff
@@ -145,6 +145,51 @@ int arch_float_enable(struct k_thread *thread, unsigned int options)
 }
 #endif /* CONFIG_FPU && CONFIG_FPU_SHARING */
 
+
+#if defined(CONFIG_XTENSA_LAZY_HIFI_SHARING)
+void xtensa_hifi_disown(struct k_thread *thread)
+{
+	unsigned int cpu_id = 0;
+	struct k_thread *owner;
+
+#if CONFIG_MP_MAX_NUM_CPUS > 1
+	cpu_id = thread->base.cpu;
+#endif
+
+	owner = atomic_ptr_get(&_kernel.cpus[cpu_id].arch.hifi_owner);
+
+	if (owner == thread) {
+		atomic_ptr_set(&_kernel.cpus[cpu_id].arch.hifi_owner, NULL);
+	}
+}
+#endif
+
+int arch_coprocessors_disable(struct k_thread *thread)
+{
+	bool enotsup = true;
+
+#if defined(CONFIG_FPU) && defined(CONFIG_FPU_SHARING)
+	arch_float_disable(thread);
+	enotsup = false;
+#endif
+
+#if defined(CONFIG_XTENSA_LAZY_HIFI_SHARING)
+	xtensa_hifi_disown(thread);
+
+	/*
+	 * This routine is only called when aborting a thread and we
+	 * deliberately do not disable the HiFi coprocessor here.
```

Comment:
Does this mean Hifi registers are saved for any crash message or debugger ?

Reply:
HiFi registers belonging to an aborted thread are not saved. Near as I can tell, Zephyr does not provide the means for GDB to obtain any coprocessor registers (regardless of architecture).